application/src/main/java/org/thingsboard/server/service/queue/DefaultTbClusterService.java

```diff
@@ -159,6 +159,10 @@ private TbMsg transformMsg(TbMsg tbMsg, DeviceProfile deviceProfile) {
             if (targetRuleChainId != null && !targetRuleChainId.equals(tbMsg.getRuleChainId())) {
                 tbMsg = TbMsg.transformMsg(tbMsg, targetRuleChainId);
             }
+            String targetQueueName = deviceProfile.getDefaultQueueName();
```

Comment:
Please add one more option and the following method

```
TbMsg.transformMsg(tbMsg, targetRuleChainId, targetQueueName)
```

Reply:
done